### ISSUE TITLE
论文里写的是用dino v2做预训练，但是代码里写的好像是用dino v1来预训练的？

### ISSUE THREAD

**Issue by gaofan3** [shermanlian/IC-ViT#5]
感谢开源！我阅读代码和论文的时候，发现论文里写的是用dino v2做预训练，但是代码里写的好像是用dino v1来预训练的？想和您确认一下

**Comment by gaofan3** [comment]
哦哦，好像论文最终版本改成是dino v1来预训练了，想问下是后面从dino v2换成dino v1了吗？还是开始的预印本版本笔误了

**Comment by shermanlian** [comment]
hi 我们是用的dino v1。 coauthor一键改成了v2。 后来我发现后就改回去了。

**Comment by shermanlian** [comment]
v1和v2效果应该差不多 训练代码也相似

**Comment by gaofan3** [comment]
感谢您的回答。另外还想请教一个问题。我看到readme里预训练的命令用的是so2sat数据集，finetune的命令用的是jumpcp数据集。我目前已经用so2sat数据集预训练完成了，然后想用so2sat的数据集微调，在您给的命令的基础上做了修改：

PYTHONPATH=/workspace/IC-ViT python main/main_supervised.py \
    trainer.devices=2 \
    trainer.max_epochs=100 \
    meta_arch/backbone=icvit_small \
    meta_arch.backbone.args.in_chans=18 \
    meta_arch.backbone.args.patch_size=8 \
    +meta_arch.backbone.args.image_size=32 \
    meta_arch.backbone.checkpoint=../ICVIT-pretrain/checkpoints/checkpoint0200_converted.pth \
    meta_arch.target='label' \
    meta_arch.num_classes=17 \
    data@train_data=so2sat_hard \
    data@val_data_dict=[so2sat_hard_val] \
    train_data.so2sat_hard.loader.batch_size=32 \
    transformations@train_transformations=so2sat \
    transformations@val_transformations=so2sat

主要是修改了一下数据集的路径，以及数据集的信息（例如图像大小是32，类别数17），以及加载的配置文件做了对应的修改。可是这样finetune之后在测试集上的识别率只有63%，和论文中的67.10%还有较大差距。

<img width="471" height="99" alt="Image" src="https://github.com/user-attachments/assets/2977e1bf-af34-4273-b647-759a5da9cc90" />

想请您看下我的微调命令有没有问题？如果有问题，能否提供一下你们在so2sat数据集上做微调的命令？非常感谢

**Comment by shermanlian** [comment]
不是这个checkpoint，这个checkpoint的代码是我之前看清理后的代码跑不跑的通的时候，随便测试的，你看下Google drive里的checkpoint可以用不，200个epoch大概率还没有收敛。论文的表格里应该是在test set上的，选的是val set上最高的epoch的test acc.

**Comment by shermanlian** [comment]
pyyhon main_supervised.py 
trainer.devices=1 
trainer.max_epochs=100 
meta_arch/backbone=icvit_small 
meta_arch.backbone.args.in_chans=18 
meta_arch.backbone.args.patch_size=8 
meta_arch.backbone.checkpoint=../checkpoints/checkpoint0100_found.pth 
meta_arch.target=label 
meta_arch.num_classes=17 
data@train_data=so2sat_hard 
data@val_data_dict=[so2sat_hard_val,so2sat_hard_test] 
train_data.so2sat_hard.loader.batch_size=32 
transformations@train_transformations=so2sat_hard 
transformations@val_transformations=so2sat_hard

**Comment by shermanlian** [comment]
能看下你的loss和acc的图吗

**Comment by gaofan3** [comment]
> 能看下你的loss和acc的图吗

<img width="1076" height="366" alt="Image" src="https://github.com/user-attachments/assets/63882a40-31eb-4db2-aef3-b38839009cd0" />

我后来发现问题了，一个是我开始微调的命令写成了transformations@train_transformations=so2sat，其实应该是so2sat_hard。另一个是main_supervised.py没能保存验证集上最高准确率的那个checkpoint，而是每个epoch后都更新保存的模型参数。而从val上的loss和acc（上图）可以看到，在验证集上的loss在上升，acc在下降，可能是因为so2sat的训练集和验证集，测试集分别是不同城市的图像，因此可能过拟合？多次实验（用您提供的预训练权重和我自己预训练的权重）表明，微调5个epoch后的效果是最好的。我修改了一下和论文的结果能对上了，修改如下：

    callbacks = [
        LearningRateMonitor(logging_interval="step"),
        # ModelCheckpoint(dirpath=cfg.trainer.default_root_dir, save_top_k=1),
        ModelCheckpoint(
            dirpath=cfg.trainer.default_root_dir, 
            save_top_k=1,
            monitor="val/acc",  # <-- 明确告诉它监控 val/acc
            mode="max",         # <-- 明确告诉它保存最大值
            filename="{epoch}-{step}-{valacc:.4f}" 
        ),
    ]

**Comment by gaofan3** [comment]
> 不是这个checkpoint，这个checkpoint的代码是我之前看清理后的代码跑不跑的通的时候，随便测试的，你看下Google drive里的checkpoint可以用不，200个epoch大概率还没有收敛。论文的表格里应该是在test set上的，选的是val set上最高的epoch的test acc.

所以一般您预训练多少个epoch呢？论文4.1节提到：
Please refer to the Supplementary Materials for additional details on datasets, training settings, and results.
但我没找到这个Supplementary Materials，能否麻烦您提供一份呢？非常感谢~！

**Comment by shermanlian** [comment]
我看了下，我好像也才pretrain了100个epoch（不好意思 记错了 不知道为什么印象中跑了500个。Supplementary Materials我看看能不能更新到arxiv上，里面也没啥，就和channelvit那一篇设置差不多。

> 我后来发现问题了，一个是我开始微调的命令写成了transformations@train_transformations=so2sat，其实应该是so2sat_hard。

是的

> 另一个是main_supervised.py没能保存验证集上最高准确率的那个checkpoint，而是每个epoch后都更新保存的模型参数。

没错 我也发现了，但是我没有用到finetune后的checkpoint 然后就没改

> 而从val上的loss和acc（上图）可以看到，在验证集上的loss在上升，acc在下降，可能是因为so2sat的训练集和验证集，测试集分别是不同城市的图像，因此可能过拟合？多次实验（用您提供的预训练权重和我自己预训练的权重）表明，微调5个epoch后的效果是最好的。

差不多的结论。我跑了三次，差不多也是这样。应该是过拟合了，我本来想找一下别的方法在这个数据集上的训练plots，但是我没找到，不知道是方法的缺陷还是数据集本身不太好训。我把三次最好的val acc的epoch对应的test acc 的平均report在论文里。

<img width="301" height="683" alt="Image" src="https://github.com/user-attachments/assets/1b6e29a6-4660-46ed-920d-d11238a58b2a" />

**Comment by shermanlian** [comment]
我还试过一个事情，但是我没有写在论文里，就是除了so2sat，我还找了两个数据集也是remote sensing，然后差不多大小的，三个数据集一起pretrain，然后finetune在so2sat上，虽然最后的test acc也是67%左右，也许高一点点，反正应该差不多，但是不会overfitting很严重，那个plots我找不到了。

**Comment by gaofan3** [comment]
> 我还试过一个事情，但是我没有写在论文里，就是除了so2sat，我还找了两个数据集也是remote sensing，然后差不多大小的，三个数据集一起pretrain，然后finetune在so2sat上，虽然最后的test acc也是67%左右，也许高一点点，反正应该差不多，但是不会overfitting很严重，那个plots我找不到了。

挺神奇的，可能只用so2sat的训练集做预训练的规模不够大？或许体现了数据规模对自监督预训练的重要性。
非常感谢您耐心的解答，我会在自己的数据集上实验一下，如果效果好（能水论文），一定给您贡献引用量！！
